# Q1: Import numpy library

In [2]:
# write your code here ^_^
import numpy as np

# Q2: Generate a sequence of 15 floats using linspace() function

In [3]:
# write your code here ^_^
np.linspace(0,15,dtype=float)

array([ 0.        ,  0.30612245,  0.6122449 ,  0.91836735,  1.2244898 ,
        1.53061224,  1.83673469,  2.14285714,  2.44897959,  2.75510204,
        3.06122449,  3.36734694,  3.67346939,  3.97959184,  4.28571429,
        4.59183673,  4.89795918,  5.20408163,  5.51020408,  5.81632653,
        6.12244898,  6.42857143,  6.73469388,  7.04081633,  7.34693878,
        7.65306122,  7.95918367,  8.26530612,  8.57142857,  8.87755102,
        9.18367347,  9.48979592,  9.79591837, 10.10204082, 10.40816327,
       10.71428571, 11.02040816, 11.32653061, 11.63265306, 11.93877551,
       12.24489796, 12.55102041, 12.85714286, 13.16326531, 13.46938776,
       13.7755102 , 14.08163265, 14.3877551 , 14.69387755, 15.        ])

# Q3: Create a 3-D array in shape (2, 2, 3) containing the four arrays given below 
Note : the array's name is up to you.

|      |       |          |         |
|------| ----- | -------- | ------- |
|arr1 | 1.1    | 2.1       | 3.1        |
|arr2 | 4.1    | 5.1       | 6.1        |
|arr3 | 7.1    | 8.1       | 9.1        |
|arr4 | 10.1   | 11.1      | 12.1       |

In [4]:
# write your code here ^_^
DArray3 = np.arange(1.1,13).reshape((2,2,3))
DArray3

array([[[ 1.1,  2.1,  3.1],
        [ 4.1,  5.1,  6.1]],

       [[ 7.1,  8.1,  9.1],
        [10.1, 11.1, 12.1]]])

# Q4: Print the following:
Note: use the same array from Q3.
- Array's type.
- Array's elements datatype.
- Array's shape.
- Array's size.
- Array's dimention.

In [5]:
# write your code here ^_^
print(type(DArray3))
print(DArray3.dtype)
print(DArray3.shape)
print(DArray3.size)
print(DArray3.ndim)

<class 'numpy.ndarray'>
float64
(2, 2, 3)
12
3


# Q5:  Change the array dimention from 3-D to 4-D
Note: use the same array from Q3.
- Create a new array to hold the changes. 
- Print the new array's dimention and shape.

In [6]:
# write your code here ^_^
DArray4 = DArray3.reshape((2,3,1,2))
print(DArray4.shape)
print(DArray4.ndim)
print(DArray4)

(2, 3, 1, 2)
4
[[[[ 1.1  2.1]]

  [[ 3.1  4.1]]

  [[ 5.1  6.1]]]


 [[[ 7.1  8.1]]

  [[ 9.1 10.1]]

  [[11.1 12.1]]]]


# Q6: Change the array's elements datatype to integer  
Note: use the same array from Q5.

- Create a new array to hold the changes. 
- Print the new array.

In [7]:
# write your code here ^_^
newD4Array = DArray4.copy()
newD4Array = newD4Array.astype(int)
print(newD4Array.dtype)
print(newD4Array)


int32
[[[[ 1  2]]

  [[ 3  4]]

  [[ 5  6]]]


 [[[ 7  8]]

  [[ 9 10]]

  [[11 12]]]]


# Q7: Print all array's elements using for loop
Note: use the same array from Q6.

Hint: use nditer()

In [8]:
# write your code here ^_^
for item in np.nditer(newD4Array):
    print(item)

1
2
3
4
5
6
7
8
9
10
11
12


# Q8:  Print number 8 using array slicing
Note: use the same array from Q6.

In [20]:
# write your code here ^_^
newD4Array[1,0,0,1]

8

# Q9: Print number 5 and number 6 using array slicing
Note: use the same array from Q6.

In [21]:
# write your code here ^_^
newD4Array[0,2,0,:]

array([5, 6])

# Q10: Search for number 8 using where()
Note: use the same array from Q6.

Note: where() is only used with small data.

*the output represents the path of the index that leads to number 8*

In [24]:
# write your code here ^_^
np.where(newD4Array == 8)

(array([1], dtype=int64),
 array([0], dtype=int64),
 array([0], dtype=int64),
 array([1], dtype=int64))

# Q11: Reshape the array as the following

    array([[[ 1, 2],
            [ 3,  4],
            [ 5,  6]],

           [[ 7,  8],
            [ 9, 10],
            [11, 12]]])
            
 Note: use the same array from Q6.

In [27]:
# write your code here ^_^
newD4Array.reshape((2,3,2))

array([[[ 1,  2],
        [ 3,  4],
        [ 5,  6]],

       [[ 7,  8],
        [ 9, 10],
        [11, 12]]])

# Q12: Join the given arrays below 
    arr1 = np.array([['A', 'B'], ['E', 'F']])
    arr2 = np.array([['C', 'D'], ['G', 'H']])
## Q12.1: Join the arrays without specifying the axis 

In [30]:
# write your code here ^_^
arr1 = np.array([['A', 'B'], ['E', 'F']])
arr2 = np.array([['C', 'D'], ['G', 'H']])

Array12 = np.concatenate((arr1,arr2))
Array12

array([['A', 'B'],
       ['E', 'F'],
       ['C', 'D'],
       ['G', 'H']], dtype='<U1')

# Q12.2: Join the arrays along rows with axis = 1 

In [31]:
# write your code here ^_^
Array13 = np.concatenate((arr1,arr2), 1)
Array13

array([['A', 'B', 'C', 'D'],
       ['E', 'F', 'G', 'H']], dtype='<U1')

# Q13: Split the array into two arrays with axis = 1, each array should contain four arrays. 
Note: use the same array from Q12.1

In [37]:
# write your code here ^_^
[A1,A2] = np.split(Array12,2, axis=1)

In [40]:
A1

array([['A'],
       ['E'],
       ['C'],
       ['G']], dtype='<U1')

In [41]:
A2

array([['B'],
       ['F'],
       ['D'],
       ['H']], dtype='<U1')